python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -320,10 +320,17 @@ def index_add(output_array, input_array, index):
 def int_array_like(int_list: List[int], like):
     """
     Converts the input list of int to a numpy array or torch tensor
-    based on the type of `like`.
+    based on the device of `like`.
+
+    If the backend is torch and the device is "meta",
+    the device is set to "cpu".
```

Comment:
could you explain why this is necessary?

Reply:
Otherwise in some cases some Labels are created on the meta device (following the fact that the corresponding block is on meta) and that gives an error

Reply:
can we give existing `labels.values` instead of the `block.values` to be the `like` argument?

Reply:
Then could you add a comment saying that we use meta in tests and that it makes this function fails in this case?